### ISSUE TITLE
[ion-datetime / ion-picker] Picker Does Not Inherit Mode Attribute from DateTime

### ISSUE THREAD

**Issue by bryplano** [ionic-team/ionic-v3#969]
### I'm submitting a ...
[x] Bug report
[  ] feature request

### Current Behavior

When setting the `mode` for an `ion-datetime` to `md` (for Material Design layout), the mode is properly applied to the `datetime` component.

However, when tapping on the `datetime`, the `picker` that appears from the datetime does not inherit the `mode` specified).

This issue occurs on iOS and Android (assuming you use `mode="md"` when running on iOS and `mode="ios"` when running on Android).

### Expected Behavior

The `picker` should inherit the `mode` attribute from the `datetime` component, if specified. 

**Example**: On iOS, if you specify `mode="md"` in the `ion-datetime` component, the `picker` should be `picker-md` when displayed.

### Reproduction Steps

See the [sample application README.md](https://github.com/bryplano/dateTimePickerV3/blob/master/README.md) (link to repo below) for instructions.

### Related Code

[Sample Application Repo](https://github.com/bryplano/dateTimePickerV3)

*Example*: On iOS, using the following component in your HTML:

```
<ion-content padding>

<p>Here is a date time component</p>

<ion-item>
  <ion-label>Date</ion-label>
  <ion-datetime mode="md"></ion-datetime>
</ion-item>

</ion-content>
```

Results in a `datetime` component that is `datetime-md` but the `picker` provided is still `picker-ios`.

![ion_picker_issue_v3](https://user-images.githubusercontent.com/39273819/52864244-5f967380-30ff-11e9-8687-601def5d6443.png)

### Other Information

Similar GH issues: #796 

**Potential Trouble Area(s) in Code**:

* https://github.com/ionic-team/ionic-v3/blob/master/src/components/picker/picker-component.ts#L61

* https://github.com/ionic-team/ionic-v3/blob/master/src/components/ion.ts#L99-L112

* https://github.com/ionic-team/ionic-v3/blob/master/src/components/datetime/datetime.ts#L518-L558

### Ionic Info Output

```
Ionic:

   ionic (Ionic CLI)  : 4.10.2
   Ionic Framework    : ionic-angular 3.9.3
   @ionic/app-scripts : 3.2.1

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 7.1.4, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 3.1.2, (and 4 other plugins)

System:

   Android SDK Tools : 26.1.1
   ios-deploy        : 2.0.0
   NodeJS            : v10.15.1 (/usr/local/bin/node)
   npm               : 6.7.0
   OS                : macOS Mojave
   Xcode             : Xcode 10.1 Build version 10B61
```

**Comment by bryplano** [comment]
Looking into this issue and it's relation to #796. 

In theory, `dateTime` should accept `pickerOptions` just like in an Ionic v4 application (rather than specify `mode=""` in the `datetime` declaration itself). 

Will report back.

**Comment by liamdebeasi** [comment]
Resolved with #974
